src/ocsp.c

```diff
@@ -1068,6 +1068,43 @@ int wolfSSL_i2d_OCSP_CERTID(WOLFSSL_OCSP_CERTID* id, unsigned char** data)
     return id->rawCertIdSize;
 }
 
+WOLFSSL_OCSP_CERTID* wolfSSL_d2i_OCSP_CERTID(WOLFSSL_OCSP_CERTID** cidOut,
+                                             const unsigned char** derIn,
+                                             int length)
+{
+    WOLFSSL_OCSP_CERTID *cid = NULL;
+
+    if ((cidOut == NULL) || (derIn == NULL) || (length == 0))
+        goto err;
+
+    cid = *cidOut;
+    /* If a NULL is passed we allocate the memory for the caller. */
+    if (!cid) {
+        cid = (WOLFSSL_OCSP_CERTID*)XMALLOC(sizeof(*cid), NULL, DYNAMIC_TYPE_OPENSSL);
+        if (!cid) goto err;
```

Comment:
Please refactor to not use goto. We discourage its use except in some places where it already exists in the code.